Bottom: 57d2df0fc68bc1cc8686912a29a17cf6b1bc6881
Top:    276082263b88ee01118b5d5289df296c18294ac1
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-01 12:20:03 +0100

Refresh of action-c-fix-the-setvalue-text

---

diff --git a/src/action.c b/src/action.c
index 14f23de..8a1e357 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2173,6 +2173,7 @@ ActionSetValue (int argc, char **argv, Coord x, Coord y)
   char *units = ARG (2);
   bool absolute;			/* flag for 'absolute' value */
   double value;
+  int text_scale;
   int err = 0;
 
   if (function && val)
@@ -2217,8 +2218,10 @@ ActionSetValue (int argc, char **argv, Coord x, Coord y)
 
 	case F_Text:
 	case F_TextScale:
-	  value /= FONT_CAPHEIGHT;
-	  SetTextScale (absolute ? value : value + Settings.TextScale);
+	  text_scale = value / (double)FONT_CAPHEIGHT * 100.;
+	  if (!absolute)
+	    text_scale += Settings.TextScale;
+	  SetTextScale (text_scale);
 	  break;
 	default:
 	  err = 1;
